woocommerce/woocommerce-ios
# [Custom Amounts M1] Order Creation - Added custom amounts UI

Closes: #10880


## Description

With this PR we implemented the UI for added custom amounts in order creation, as well as hooking the events to their actions in the `EditableOrderViewModel`. Note as well that we re-designed the Products section, adding a header title containing the plus button when products are already added. This replaces the old approach, which leaves the Add Product button even if we have products added to the order. (See Video).

In order to keep the PR in a reasonable size we only include the add and delete custom amounts here. Editing a custom amount will be implemented in the next PR.

## Testing instructions


1. Go to orders
2. See empty state design
3. Add a product. See how the Products section is added with barcode and add button. Tests these as well.
4. Add a custom amount. See that it's added to the Order creation screen.
5. See that we have now a header for the Custom Amounts section in the same way we have for products when items are added. Tap on + to add a new custom amount and see that it's properly added.
6. Delete a custom amount by tapping on X. See that it's removed.
7. Remove the other custom amount. See that it's removed and the original empty state is shown.

## Notes
- As we've seen in our previous PR, the old Fee logic doesn't work well with this new custom amounts logic. This is fine, as they won't be existing at the same time.
- On that note, please disable the feature flag `orderCustomAmountsM1` to check that everything still works when disabled

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/8075f736-71b7-4b22-815c-9dcd38bb5ddd

---
